lib/compilers/hook.ts

```diff
@@ -52,4 +54,33 @@ export class HookCompiler extends BaseCompiler {
         options.push(outputFilename);
         return super.runCompiler(compiler, options, inputFilename, execOptions);
     }
+
+    override processAsm(result) {
```

Comment:
Would appreciate it if you could add some tests for this :)

Reply:
Sure. I'll do.

Reply:
Are the compiler tests in that `test/compilers` folder? I just found a `clang-tests.js` in there.

Reply:
Not much testing is done for the compilers themselves which is what we're trying to change, yes :) But feel free to create the tests wherever makes more sense to you

Reply:
@RubenRBS, I covered the lines that this PR touched on. See if that's enough, please.